Work/create_x_queues.sh

```diff
@@ -16,38 +41,50 @@ do
         shift
         shift
         ;;
-        *)
+## Added another option to set the size of BATCH
+        --batch)
+	BATCH=$2
+	shift
+	shift
+	;;
+	*)
         POS+=("$1")
         shift
         ;;
     esac
 done
+
 set -- "${POS[@]}"
 
 COUNT=$1
+START=0
+COUNTER=0
+
+## If you accidentaly use bigger BATCH than # of servers
+[[ $COUNT -lt $BATCH ]] && BATCH=$COUNT
 
 for (( i=0; i < ${COUNT}; i++ )) 
 do
+    (( COUNTER++ ))
+    if [[ $COUNTER -eq $BATCH ]];then
+        openshift $START $i &
+        START=$(( i + 1 ))
+        COUNTER=0
+    fi
+done
 
-  if [[ "${DEL}" -eq 1 ]]; then
-     oc delete address seantest-addr.${STEM}${i}   # can change this value to correct 'short' addr space name ( and value below in metadata
-  else
+## Run the rest of servers which didn't fit into last whole BATCH
+[[ $COUNTER != 0 ]] && openshift $START $(( COUNT - 1 )) &
 
-      oc --as=developer create -f - << EOF
-apiVersion: enmasse.io/v1alpha1
-kind: Address
-metadata:
-    name: seantest-addr.${STEM}.${i}
-spec:
-    address: ${STEM}.${i}
-    type: queue
-    plan: pooled-queue
-EOF
-
-  fi
 
+## The waiting loop untill all processes in background finish (2 because grep will show also itself in ps)
+Name=$(basename $0)
+while [[ $(ps -e |grep -c -i $NAME) -gt 2 ]];do
```

Comment:
grep is failing here, I suppose its because the NAME variable is different to Name being set in the line above